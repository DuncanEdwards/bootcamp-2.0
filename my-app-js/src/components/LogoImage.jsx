export const LogoImage = ({ className, logo, alt }) => {
  return <img src={logo} className={className} alt={alt} />;
};
