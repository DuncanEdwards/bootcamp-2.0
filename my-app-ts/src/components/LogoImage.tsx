type LogoImageProps = {
  logo: string;
  alt: string;
  className?: string;
};

export const LogoImage = ({
  alt,
  className = "App-logo",
  logo,
}: LogoImageProps) => {
  return <img src={logo} className={`logo-image ${className}`} alt={alt} />;
};
