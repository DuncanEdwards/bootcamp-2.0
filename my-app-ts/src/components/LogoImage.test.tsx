/* eslint-disable testing-library/no-debugging-utils */
import { LogoImage } from "./LogoImage";
import { render, screen } from "@testing-library/react";

test("LogoImage with all props provide", () => {
  render(
    <LogoImage
      logo={"test-extra.jpg"}
      alt="logo image"
      className="my-classname"
    />
  );

  const image = screen.getByRole("img");
  expect(image).toMatchSnapshot();
});

test("LogoImage without className provided", () => {
  render(<LogoImage logo={"test-extra.jpg"} alt="logo image" />);

  const image = screen.getByRole("img");
  expect(image).toMatchSnapshot();
});
