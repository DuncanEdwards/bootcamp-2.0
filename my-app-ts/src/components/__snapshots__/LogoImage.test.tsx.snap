// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoImage with all props provide 1`] = `
<img
  alt="logo image"
  class="my-classname"
  src="test-extra.jpg"
/>
`;

exports[`LogoImage without className provided 1`] = `
<img
  alt="logo image"
  class="App-logo"
  src="test-extra.jpg"
/>
`;
